src/bun.js/ipc.zig

```diff
@@ -318,3 +463,157 @@ pub fn NewIPCHandler(comptime Context: type) type {
         ) void {}
     };
 }
+
+fn NewNamedPipeIPCHandler(comptime Context: type) type {
+    const uv = bun.windows.libuv;
+    return struct {
+        fn uvStreamAllocCallback(handle: *uv.uv_handle_t, suggested_size: usize, buffer: *uv.uv_buf_t) callconv(.C) void {
+            const this: *Context = @ptrCast(@alignCast(handle.data));
+
+            var available = this.ipc.incoming.available();
+            if (available.len < suggested_size) {
+                this.ipc.incoming.ensureUnusedCapacity(bun.default_allocator, suggested_size) catch bun.outOfMemory();
+                available = this.ipc.incoming.available();
+            }
+            log("uvStreamAllocCallback {d}", .{suggested_size});
+            buffer.* = .{ .base = @ptrCast(available.ptr), .len = @intCast(suggested_size) };
+        }
+
+        fn uvStreamReadCallback(handle: *uv.uv_handle_t, nread: isize, buffer: *const uv.uv_buf_t) callconv(.C) void {
```

Comment:
This should get a small wrapper, and the signature should look something like:

```suggestion
        pub fn onRead(this: *Context, buffer: []const u8) void {
```